iceoryx_utils/test/moduletests/test_cxx_list.cpp

```diff
@@ -0,0 +1,2283 @@
+    }
+}
+
+TEST_F(list_test, PushBackCheckInsertPosition)
+{
+    uint64_t i = 0;
+
+    for (; i < TESTLISTCAPACITY; ++i)
+    {
+        EXPECT_TRUE(sut.push_back(i));
+    }
+
+    i = 0;
+    for (auto& listElement : sut)
+    {
+        EXPECT_THAT(listElement.m_value, Eq(i));
+        ++i;
+    }
+}
+
+TEST_F(list_test, AccessFrontElement)
+{
+    constexpr int64_t DEFAULT_VALUE{13};
+    const TestListElement a{DEFAULT_VALUE};
+
+    sut.push_front({});
+    sut.push_front(a);
+
+    TestListElement& b{sut.front()};
+    const TestListElement& c{sut.front()};
+    EXPECT_THAT(b.m_value, Eq(DEFAULT_VALUE));
+    EXPECT_THAT(c.m_value, Eq(DEFAULT_VALUE));
+}
+
+TEST_F(list_test, AccessFrontElementFromConstList)
+{
+    constexpr int64_t DEFAULT_VALUE{13};
+    const TestListElement a{DEFAULT_VALUE};
+
+    sut.push_front({});
+    sut.push_front(a);
+
+    const list<TestListElement, TESTLISTCAPACITY> sut1{sut};
+    const TestListElement& c = sut1.front();
+
+    EXPECT_THAT(c.m_value, Eq(DEFAULT_VALUE));
+}
+
+TEST_F(list_test, AccessBackElement)
+{
+    constexpr int64_t DEFAULT_VALUE{13};
+    const TestListElement a{DEFAULT_VALUE};
+
+    sut.push_front(a);
+    sut.push_front({});
+
+    TestListElement& b{sut.back()};
+    const TestListElement& c{sut.back()};
+    EXPECT_THAT(b.m_value, Eq(DEFAULT_VALUE));
+    EXPECT_THAT(c.m_value, Eq(DEFAULT_VALUE));
+}
+
+TEST_F(list_test, AccessBackElementFromConstList)
+{
+    constexpr int64_t DEFAULT_VALUE{13};
+    const TestListElement a{DEFAULT_VALUE};
+
+    sut.push_front(a);
+    sut.push_front({});
+
+    const list<TestListElement, TESTLISTCAPACITY> sut1{sut};
+    const TestListElement& c = sut1.back();
+
+    EXPECT_THAT(c.m_value, Eq(DEFAULT_VALUE));
+}
+
+TEST_F(list_test, PopFrontOnEmptyList)
+{
+    EXPECT_FALSE(sut.pop_front());
+    ASSERT_THAT(sut.size(), Eq(0u));
+    EXPECT_THAT(isSetupState(), Eq(true));
+}
+
+TEST_F(list_test, PopFrontNonEmptyList)
+{
+    sut.emplace_front(101);
+    ASSERT_THAT(sut.size(), Eq(1u));
+
+    EXPECT_TRUE(sut.pop_front());
+
+    ASSERT_THAT(sut.size(), Eq(0u));
+    ASSERT_THAT(cTor, Eq(0u));
+    EXPECT_THAT(customCTor, Eq(1u));
+    ASSERT_THAT(dTor, Eq(1u));
+}
+
+TEST_F(list_test, PopFrontFullToEmptyList)
+{
+    for (uint64_t i = 0; i < TESTLISTCAPACITY; ++i)
+    {
+        sut.emplace_front();
+        EXPECT_THAT(sut.size(), Eq(i + 1));
+    }
+
+    for (uint64_t i = 0; i < TESTLISTCAPACITY; ++i)
+    {
+        EXPECT_THAT(sut.size(), Eq(TESTLISTCAPACITY - i));
+        EXPECT_TRUE(sut.pop_front());
+    }
+
+    ASSERT_THAT(sut.size(), Eq(0u));
+    ASSERT_THAT(cTor, Eq(TESTLISTCAPACITY));
+    ASSERT_THAT(dTor, Eq(TESTLISTCAPACITY));
+}
+
+TEST_F(list_test, PopFrontFullPlusOneToEmptyList)
+{
+    for (uint64_t i = 0; i < TESTLISTCAPACITY; ++i)
+    {
+        sut.emplace(sut.cbegin());
+        EXPECT_THAT(sut.size(), Eq((i + 1) > TESTLISTCAPACITY ? TESTLISTCAPACITY : (i + 1)));
```

Comment:
Does not matter much here but normally it is better to add an extra statement after the loop for the final emplace to eliminate the branch in the loop which is only needed in the final iteration.

Does not have to be changed though. Just an observation which is important for performance  and arguably readability. (The reader may think on first sight the condition could be true every iteration if it would be more complex.)

Reply:
Background: this implementation code stems from a parametrized test case (w/ list filling as parameter). 
The test implementation being however does not use parametrized tests (yet).